website/docs/usage/large-language-models.mdx

```diff
@@ -0,0 +1,495 @@
+  functionality to parse the resulting response back into structured fields on
+  spaCy's [Doc](https://spacy.io/api/doc) objects.
+- A [**model**](#models) defining the model to use and how to connect to it.
+  Note that `spacy-llm` supports both access to external APIs (such as OpenAI)
+  as well as access to self-hosted open-source LLMs (such as using Dolly through
+  Hugging Face).
+
+Moreover, `spacy-llm` exposes a customizable [**caching**](#cache) functionality
+to avoid running the same document through an LLM service (be it local or
+through a REST API) more than once.
+
+Finally, you can choose to save a stringified version of LLM prompts/responses
+within the `Doc.user_data["llm_io"]` attribute by setting `save_io` to `True`.
+`Doc.user_data["llm_io"]` is a dictionary containing one entry for every LLM
+component within the spaCy pipeline. Each entry is itself a dictionary, with two
+keys: `prompt` and `response`.
+
+A note on `validate_types`: by default, `spacy-llm` checks whether the
+signatures of the `model` and `task` callables are consistent with each other
+and emits a warning if they don't. `validate_types` can be set to `False` if you
+want to disable this behavior.
+
+### Tasks {id="tasks"}
+
+A _task_ defines an NLP problem or question, that will be sent to the LLM via a
+prompt. Further, the task defines how to parse the LLM's responses back into
+structured information. All tasks are registered in spaCy's `llm_tasks`
+registry.
+
+Practically speaking, a task should adhere to the `Protocol` `LLMTask` defined
+in [`ty.py`](https://github.com/spacy-llm/spacy_llm/ty.py). It needs to define a
+`generate_prompts` function and a `parse_responses` function.
+
+Moreover, the task may define an optional
+[`scorer` method](https://spacy.io/api/scorer#score). It should accept an
+iterable of `Example`s as input and return a score dictionary. If the `scorer`
+method is defined, `spacy-llm` will call it to evaluate the component.
+
+| Component                                                                   | Description                                                                                                                                                           |
+| --------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`task.generate_prompts`](/api/large-language-models#task-generate-prompts) | Takes a collection of documents, and returns a collection of "prompts", which can be of type `Any`.                                                                   |
+| [`task.parse_responses`](/api/large-language-models#task-parse-responses)   | Takes a collection of LLM responses and the original documents, parses the responses into structured information, and sets the annotations on the documents.          |
```

Comment:
The table would look nicer if we can have these first two methods somewhere else, maybe inbetween the previous 2 paragraphs?